### PR TITLE
Add SS58 Registry

### PR DESCRIPTION
**Pull request by joepetrowski**
Closes https://github.com/paritytech/substrate/issues/7013

To review:
- Data types, e.g. should prefix be `0` or `"0"`?
- Data fields, should anything be added or removed?
- Rust usability, can you automate e.g. Subkey off this file? @bkchr 
- Polkadot JS usability @jacogr 

I entered these manually off of [this code](https://github.com/paritytech/substrate/blob/a063383/primitives/core/src/crypto.rs#L435). Someone do a sanity check please.

Once we are happy with the above we can ask teams to fill out their own token/website info.

### CODE REVIEW

**Comment by jacogr** [comment]
Looks sane to me on first glance. I would probably make the numerics explicit numbers. (decimals & format). The best is actually in a completely separate repo to it can be pulled as-is, but either way, this will go a long way to solving cross-compat. (As long as it is usable here, I'll be happy and can make it work)

**Comment by joepetrowski** [comment]
I was also thinking about a separate repo. Might make sense to hold many registries, like XCM identifiers.

File: ss58-registry.json [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+			"displayName": "This prefix is reserved.",
+			"symbol": null,
+			"decimals": null,
+			"standardAccount": null,
+			"website": null
+		},
+		{
+			"prefix": "48",
+			"network": "reserved48",
+			"displayName": "All prefixes above 48 are reserved.",
```
Comment (jacogr) [inline ss58-registry.json:268]:
```suggestion
			"displayName": "All prefixes 48 and higher are reserved and cannot be allocated.",
```

The reason for this change ... at least 2 people have stepped into this trying to claim >48. My wording aims to clarify that, although it is perfectly understandable to me as-is. Then again, my wording may needs some tweaks as well...

Comment (joepetrowski) [reply ss58-registry.json:268]: No, this is definitely better.

**Comment by apopiak** [comment]
Also speaking out in favor of having the numbers (decimals & format) be represented as numbers in json.

**Review by bkchr — APPROVED** [review]
I merged one more, now I'm open to merge this. Sorry for the delay. Could you add the dark nerwork here as well?

I will take care to write a macro that parses this json and generates the correct Rust code.

Could we may get all the other data like website etc for the individual projects in here?

**Comment by jacogr** [comment]
Once your parsing is done, would like to move it into a separate repo ss58formats (crate for Rust, with package.json for JS, ... other languages?) so we can pull it in as a dependency wherever needed.

**Comment by bkchr** [comment]
Creating a separate crate was also my idea, moving it to a complete different repo also sounds fine by me.

**Comment by jacogr** [comment]
The repo (with a crate & package.json) make it easier to pull it is from non-Rust repos.

**Comment by joepetrowski** [comment]
> Could we may get all the other data like website etc for the individual projects in here?

I was thinking individual projects could make PRs for their own so I don't get the wrong one. I can get the more well known ones into this PR at least.

File: ss58-registry.json [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+			"website": null
+		},
+		{
+			"prefix": 13,
+			"network": "substratee",
+			"displayName": "SubstraTEE",
+			"symbols": null,
+			"decimals": null,
+			"standardAccount": "*25519",
+			"website": null
```
Comment (brenzi) [inline ss58-registry.json:137]: www.substratee.com

File: ss58-registry.json [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+		},
+		{
+			"prefix": 13,
+			"network": "substratee",
+			"displayName": "SubstraTEE",
+			"symbols": null,
```
Comment (brenzi) [inline ss58-registry.json:134]: as substraTEE is a framework, we do not define a token symbol (nor decimals).

File: ss58-registry.json [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+			"decimals": null,
+			"standardAccount": "*25519",
+			"website": null
+		},
+		{
+			"prefix": 36,
+			"network": "centrifuge",
+			"displayName": "Centrifuge Chain",
+			"symbols": ["RAD"],
+			"decimals": null,
```
Comment (vedhavyas) [inline ss58-registry.json:243]:
```suggestion
			"decimals": [18],
```

File: ss58-registry.json [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+			"standardAccount": "*25519",
+			"website": null
```
Comment (vedhavyas) [inline ss58-registry.json:245]:
```suggestion
			"website": "https://centrifuge.io/"
```

File: ss58-registry.json (hunk 2 above)
Comment (jacogr) [inline ss58-registry.json:137]:
```suggestion
			"website": "https://www.substratee.com"
```

**Pull request merged**